src/main/java/org/broadinstitute/hellbender/utils/genotyper/AlleleLikelihoods.java

```diff
@@ -1216,18 +1189,26 @@ public void filterPoorlyModeledEvidence(final ToDoubleFunction<EVIDENCE> log10Mi
 
     private Object2IntMap<EVIDENCE> evidenceIndexBySampleIndex(final int sampleIndex) {
         if (evidenceIndexBySampleIndex.get(sampleIndex) == null) {
-            final List<EVIDENCE> sampleEvidence = evidenceBySampleIndex.get(sampleIndex);
-            final int sampleEvidenceCount = sampleEvidence.size();
-            final Object2IntMap<EVIDENCE> index = new Object2IntOpenHashMap<>(sampleEvidenceCount);
-            index.defaultReturnValue(MISSING_INDEX);
-            evidenceIndexBySampleIndex.set(sampleIndex, index);
-            for (int r = 0; r < sampleEvidenceCount; r++) {
-                index.put(sampleEvidence.get(r), r);
-            }
-            return index;
-        } else {
-            return evidenceIndexBySampleIndex.get(sampleIndex);
+            fillEvidenceToIndexCache(sampleIndex);
         }
+        return evidenceIndexBySampleIndex.get(sampleIndex);
+    }
+
+    @VisibleForTesting
+    void fillEvidenceToIndexCache(int sampleIndex) {
+        final List<EVIDENCE> sampleEvidence = evidenceBySampleIndex.get(sampleIndex);
+        final int sampleEvidenceCount = sampleEvidence.size();
+        final Object2IntMap<EVIDENCE> index = new Object2IntOpenHashMap<>(sampleEvidenceCount);
+        index.defaultReturnValue(MISSING_INDEX);
+        evidenceIndexBySampleIndex.set(sampleIndex, index);
+        for (int r = 0; r < sampleEvidenceCount; r++) {
```

Comment:
what about using a ```foreach```:

```
int nextIdx = 0;
for( final EVIDENCE evi : sampleEvidence) {
    index.put(evi, nextIdx++);
}
```

Reply:
I like the `foreach` but my distaste for introducing a variable outside the scope of the `for` loop is stronger.